galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/ICertificateStoreService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework.spi;
+
+
+public interface ICertificateStoreService {
+	
+	/**
+	 * <p>
+	 * Retrieves the PEM certificate as a string.
+	 * </p>
+	 * 
+	 * @param id
+	 * @return String certificate
+	 * @throws CertificateStoreException if no certificate found
+	 */
+	public String getPem(String id) throws CertificateStoreException;
```

Comment:
Is throwing an exception if the cert is not found the right approach here?  Feels a bit drastic

Reply:
Yeah was 50/50 about just returning an empty string. But if we are searching for an explicit cert especially from a Manager then i thought this could be cause for an exception (with related message)

Reply:
Throwing an exception is consistent with everything else,   eg creds.

Reply:
needs @NotNull n parameter

Reply:
Infact,  does not need to be getPem or getDer.    Just needs to be getX509Certificate.     The store will look for .pem or .der and load the approriate format.
Needs to return as a certificate,  not a string, ie binary x509 certificate.